### PR TITLE
docs: document CEL `in` operator and `exists()` map behavior with tests

### PR DESCRIPTION
**Pull request by ChathurangaKCD**
## Checklist
- [ ] Tests added or updated (unit, integration, etc.)
- [ ] Samples updated (if applicable)

This PR documents CEL `in` operator and `exists()` map behavior and adds tests to cover those cases.

### Files Changed (by top-level folder)
- docs/: 1 file changed, +22 lines
- internal/: 1 file changed, +55 lines
- Total: 2 files changed, +77 lines, 0 deletions

### Changes Summary
- docs/templating/templating.md (+22)
  - Added "Membership and Existence Checks" under CEL Capabilities > Map Access Notation.
  - Documents `in` operator for:
    - Map key membership (e.g., `"key" in map`)
    - Primitive list membership (e.g., `"item" in list`)
  - Documents `exists()` and `all()` behavior when iterating map keys and recommends bracket notation for value access with examples.
- internal/template/engine_test.go (+55)
  - Added 3 tests to TestEngineRender:
    1. `in` operator for map key existence
    2. `in` operator for list membership
    3. `exists()` macro iterating over map keys with conditional checks
  - Each test includes template, JSON inputs, and expected YAML output.
  - Extends test coverage of CEL/templating behavior (now ~109+ tests in TestEngineRender as noted in repo context).

### API / CRD Surface Changes
- None. No API types, CRD definitions, exported/public signatures modified.
- Compatibility risk: Low (documentation and tests only)

### Tests Added / Updated
- Added: 3 new unit tests targeting templating/CEL expressions (`in`, `exists()`).
- Coverage: Core operator behaviors (`in`, `exists()`) for maps and lists are now covered.
- Remaining gaps: No tests documenting error/edge cases for unsupported operand types or type-mismatch behavior of `in`/`exists()`.

### Risk Hotspots
- AuthN/AuthZ, RBAC, secrets, reconciliation loops, install/upgrade paths: No changes → Low risk.
- Change type: Documentation + tests only — no runtime/behavioral changes.

Summary: Low-risk documentation enhancement with focused unit tests improving CEL templating docs and behavioral coverage for membership/exists operations.

### CODE REVIEW

File: docs/templating/templating.md [hunk 1]
```diff
@@ -106,6 +106,28 @@ Bracket notation is **required** for:
...
+
+The `in` operator checks membership in maps and lists:
+
+```
+# Check if a key exists in a map
+parameters.endpointName in workload.endpoints
+
+# Check if a value exists in a list (primitives only — strings, ints, bools)
+"HTTP" in parameters.allowedTypes
+```
```
[the remark this answers is not in the document]
Comment (ChathurangaKCD) [reply docs/templating/templating.md:119]: noted. will fix with other improvements to the same file.
